gobblin-api/src/main/java/gobblin/writer/StreamEncoder.java

```diff
@@ -0,0 +1,25 @@
+package gobblin.writer;
+
+import java.io.OutputStream;
+
+import gobblin.annotation.Alpha;
+
+
+/**
+ * Interface for an object that can encrypt a bytestream.
+ */
+@Alpha
+public interface StreamEncoder {
```

Comment:
I actually have a separate interface StreamDecoder sitting unchecked in - figure might as well allow them to be separate concepts